app/controllers/provider/sessions_controller.rb

```diff
@@ -22,6 +22,7 @@ def create
 
     if @user
       self.current_user = @user
+      cookies[:first_login] = { value: 'true' } if current_user.user_sessions.empty?
```

Comment:
IMO you don't need to use cookies, that's stored on the client and can be manipulated by the user. Specially when the cookie is not encrypted. If any, you should use the store session. However, I don't think you need the session neither. Here you could just store whether this is the first login:

```suggestion
      first = current_user.user_sessions.empty?
```

And then a couple lines below set the flash notice accordingly:

```ruby
flash[:notice] = first ? 'You can use quick starts to learn about 3scale features step by step. <a href="/p/admin/quickstarts">Take tour</a>'.html_safe : 'Signed in successfully'
```
Even better if you use i18n and load the text from `en.yml`